File: Demo2D/Assets/Scripts/Globals/GlobalObjectManager.cs
using UnityEngine;
using System.Collections;

public class GlobalObjectManager : MonoBehaviour {

	public Transform guiManagerPrefab;

	GameObject guiManager;
	GameObject soundManager;


	void Awake() {
		if (!(GameObject.FindWithTag("GUIMANAGER"))) {    
			Transform guiManagerInstance = (Transform)Instantiate(guiManagerPrefab, transform.position, transform.rotation);
			guiManagerInstance.name = "guiManager";
		}
	}

	void Start() {
		
		guiManager = GameObject.Find("guiManager");

	}
}

File: Demo2D/Assets/Scripts/Globals/GlobalObjectManager.js
#pragma strict

var guiManagerPrefab : Transform;
var guiManager : GameObject;
var soundManager : GameObject;


function Awake() {
  if (!(GameObject.FindWithTag("GUIMANAGER"))) {    
    var guiManagerInstance : Transform = Instantiate(guiManagerPrefab, transform.position, transform.rotation);
    guiManagerInstance.name = "guiManager";
  }
}

function Start() {
  guiManager = GameObject.Find("guiManager");
}